Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -569,14 +574,47 @@ extension RouteController: Router {
     }
 
     public func updateRoute(with indexedRouteResponse: IndexedRouteResponse, routeOptions: RouteOptions?) {
-        guard let routes = indexedRouteResponse.routeResponse.routes, routes.count > indexedRouteResponse.routeIndex else {
+        guard let routes = indexedRouteResponse.routeResponse.routes,
+              routes.count > indexedRouteResponse.routeIndex else {
             preconditionFailure("`indexedRouteResponse` does not contain route for index `\(indexedRouteResponse.routeIndex)` when updating route.")
         }
+        let route = routes[indexedRouteResponse.routeIndex]
+        if shouldStartNewBillingSession(for: route, routeOptions: routeOptions) {
+            BillingHandler.shared.stopBillingSession(with: sessionUUID)
+            BillingHandler.shared.beginBillingSession(for: .activeGuidance, uuid: sessionUUID)
+        }
+
         let routeOptions = routeOptions ?? routeProgress.routeOptions
-        routeProgress = RouteProgress(route: routes[indexedRouteResponse.routeIndex], options: routeOptions)
+        routeProgress = RouteProgress(route: route, options: routeOptions)
         self.indexedRouteResponse = indexedRouteResponse
         updateNavigator(with: routeProgress)
     }
+
+    private func shouldStartNewBillingSession(for newRoute: Route, routeOptions: RouteOptions?) -> Bool {
```

Comment:
This logic seems to be business-related. Do you have confirmation for the conditions when we do or do not bill for changing active route?

Reply:
Discussed